components/ShareScreen.js

```diff
@@ -0,0 +1,225 @@
+import React, { useState } from 'react';
+import { Share, StyleSheet, SafeAreaView, ScrollView, KeyboardAvoidingView } from 'react-native';
+import styled from 'styled-components/native';
+
+import crafty from '../assets/crafty.png';
+import sunny from '../assets/sunny.png';
+import active from '../assets/active.png';
+import smiley from '../assets/smiley.png';
+
+export const ShareScreen = () => {
+  const [message, setMessage] = useState('');
+  const [mood, setMood] = useState('');
+
+  //Function to implement the Share on phone button, the text content that will be shared is what
+  //appears in the message property on line 20. I am generating a combination of the set Mood, the phrase
+  //explaining that Mood and also the personal message that the user has entered in the TextInput field.
+  const onShare = async () => {
+    try {
+      const result = await Share.share({
+        message: `You've received a ${mood} mood booster! ${phrase()} A special message just for you: ${message}`,
+      });
+      if (result.action === Share.sharedAction) {
+        if (result.activityType) {
+          // shared with activity type of result.activityType
+        } else {
+          // shared
+        }
+      } else if (result.action === Share.dismissedAction) {
+        // dismissed
+      }
+    } catch (error) {
+      alert(error.message);
+    }
+  };
+
+  //Function to generate the phrase that will be showing on screen and shared at the end. The shown
+  //phrase will depend on the mood that has been chosen and set via the mood state
+  const phrase = () => {
+    if(mood === 'Crafty') {
+      return 'Getting creative and making something with your hands can help set the positive vibes 🧶!';
+    } else if(mood === 'Sunny') {
+      return 'Going outdoors and enjoying some fresh air is an instant mood boost ☀!';
+    } else if(mood === 'Active') {
+      return 'Move throughout the day! Breaking a sweat helps you handle stress hormones 🏃‍♀️.';
+    } else if(mood === 'Smiley') {
+      return 'Chin up and smile! Our facial expressions influence our thoughts and emotions 😁.';
+    } else {
+      return 'Set a mood booster for today!';
+    }
+  };
```

Comment:
Maybe [switch case](https://www.w3schools.com/js/js_switch.asp) could be an option here